crates/bootstrap/src/lib.rs

```diff
@@ -27,14 +28,201 @@ pub struct ExampleState {
+        // a sub-viewport with the given edge inset on all sides.
+        let inset = get_inset_override();
+        if inset.is_some() {
+            app.window_mut().set_automatic_viewport(false);
+        }
+
+        // Preload some textures for the examples to use.
+        self.monkey = Some(
+            self.yak
+                .add_texture(load_texture(MONKEY_PNG, TextureFilter::Linear)),
+        );
+        self.monkey_blurred = Some(
+            self.yak
+                .add_texture(load_texture(MONKEY_BLURRED_PNG, TextureFilter::Linear)),
+        );
+        self.brown_inlay = Some(
+            self.yak
+                .add_texture(load_texture(BROWN_INLAY_PNG, TextureFilter::Nearest)),
+        );
+        self.custom = Some(app.renderer.add_texture(
+            custom_texture::generate(&app.device, &app.queue),
+            wgpu::FilterMode::Nearest,
+            wgpu::FilterMode::Nearest,
+            wgpu::FilterMode::Nearest,
+        ));
+
+        // Add a custom font for some of the examples.
+        let fonts = self.yak.dom().get_global_or_init(Fonts::default);
+        let font = Font::from_bytes(
+            include_bytes!("../assets/Hack-Regular.ttf").as_slice(),
+            FontSettings::default(),
+        )
+        .unwrap();
+
+        fonts.add(font, Some("monospace"));
+
+        self.app = Some(app);
+        self.window = Some(window);
+    }
+
+    fn window_event(
+        &mut self,
+        event_loop: &ActiveEventLoop,
+        window_id: WindowId,
+        event: WindowEvent,
+    ) {
+        // `unwrap` is fine, the window will always be available when
+        // receiving a window event.
+
+        if self.app.as_mut().unwrap().handle_event(
+            &mut self.yak,
+            &Event::<()>::WindowEvent {
+                window_id,
+                event: event.clone(),
+            },
+            event_loop,
+        ) {
+            return;
+        }
+
+        // Handle window event.
+        match event {
+            WindowEvent::RedrawRequested => {
+                self.time = (Instant::now() - self.start).as_secs_f32();
+
+                {
+                    profiling::scope!("Build UI");
+
+                    // Every frame, call yak.start() to begin building the UI for
+                    // this frame. Any yakui widget calls that happen on this thread
+                    // between start() and finish() will be applied to this yakui
+                    // State.
+                    self.yak.start();
+
+                    // Call out to the body of the program, passing in a bit of
+                    // shared state that all the examples can use.
+                    let body = {
+                        let body = self.body.take().unwrap();
+                        body.run(self);
+
+                        body
+                    };
+
+                    self.body = Some(body);
```

Comment:
Separating the app states, mentioned above, will fix this funny borrow checker appeasement dance